internal/store/pod.go

```diff
@@ -343,6 +343,58 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_pod_status_condition",
```

Comment:
So we already have kube_pod_status_unschedulable, kube_pod_status_ready, kube_pod_status_scheduled. What other conditions would there be? This would be duplicating those metrics no?

Reply:
That was my point, the initial need was adding a new gauge kube_pod_status_ready_time with the exact same code than the existing [kube_pod_status_scheduled_time](https://github.com/kubernetes/kube-state-metrics/blob/1d132e7e728face5ba35be3f3310139c98f3a001/internal/store/pod.go#L236).

Reply:
Reworking the overall way status are exposed could definitely be discussed but I think it is out of scope of this PR.

Reply:
@lilic yes, the scope of this PR was basically adding `kube_pod_status_condition_last_transition_time`.

You may be interested in `ContainersReady` and `Initialized`, but these can be done the same way as `kube_pod_status_ready` and I think it's out of scope. I will remove `kube_pod_status_condition`.